servo-storage: rusqlite 0.37 prevents using sea-query-rusqlite 0.8.0-rc.17 (build failure with sea-query 1.0.0-rc.33)

## Description

`servo-storage` v0.1.0-rc2 depends on:
- `rusqlite` 0.37 (`libsqlite3-sys` 0.35)
- `sea-query` ^1.0.0-rc
- `sea-query-rusqlite` ^0.8.0-rc

`sea-query` 1.0.0-rc.33 (released 2026-03-18) added a `Value::Enum` variant. The matching fix in `sea-query-rusqlite` was published as 0.8.0-rc.17, but that version requires `rusqlite` 0.38 (`libsqlite3-sys` 0.36).

Because `servo-storage` pins `rusqlite` 0.37, Cargo cannot upgrade `sea-query-rusqlite` past 0.8.0-rc.15 due to the `links = "sqlite3"` native library conflict. Any fresh resolve (without a lockfile) picks up `sea-query` rc.33 paired with the old `sea-query-rusqlite` rc.15, causing:

```
error[E0004]: non-exhaustive patterns: `&sea_query::Value::Enum(_)` not covered
  --> sea-query-rusqlite-0.8.0-rc.15/src/lib.rs:67:15
```

## Suggested fix

Bump `rusqlite` from 0.37 to 0.38 in `servo-storage`, which would allow `sea-query-rusqlite` 0.8.0-rc.17 to resolve and handle the new `Value::Enum` variant.

## Impact

This breaks any downstream consumer of `servo` / `servo-storage` that does a fresh dependency resolve without a pinned lockfile. For example, the [Slint servo example](https://github.com/slint-ui/slint/tree/master/examples/servo) CI is currently broken by this.

## Related

- SeaQL/sea-query#1065 — the `sea-query` side of this issue
- [sea-query PR #1051](https://github.com/SeaQL/sea-query/pull/1051) — the PR that added `Value::Enum`

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -158,8 +158,8 @@ rustc-hash = "2.1.2"
 rustls = { version = "0.23", default-features = false, features = ["logging", "std", "tls12"] }
 rustls-pki-types = "1.14"
 rustls-platform-verifier = "0.6.2"
-sea-query = { version = "1.0.0-rc.31", default-features = false, features = ["backend-sqlite", "derive"] }
-sea-query-rusqlite = { version = "0.8.0-rc.15" }
+sea-query = { version = "=1.0.0-rc.31", default-features = false, features = ["backend-sqlite", "derive"] }
+sea-query-rusqlite = { version = "=0.8.0-rc.15" }
 sec1 = "0.7"
 selectors = { git = "https://github.com/servo/stylo", rev = "96ceb5405bd7ce10c4141c62e860d50745a75547" }
 serde = "1.0.228"

